docs/app/Examples/modules/Transition/Types/index.js

```diff
@@ -6,23 +6,30 @@ import ExampleSection from 'docs/app/Components/ComponentDoc/ExampleSection'
 
 const TransitionTypesExamples = () => (
   <ExampleSection title='Types'>
+
     <ComponentExample
       title='Transition'
       description='A Transition animates a single child by toggling the the visible prop.'
       examplePath='modules/Transition/Types/TransitionExampleTransition'
     >
       <Message warning>
-        <p>Do not unmount a Transition child or else it cannot be animated.</p>
+        <p>An animated component must not unmount unexpectedly or the transition will not run as intended.</p>
         <Message.List>
           <Message.Item>
-            Use the <code>unmountOnHide</code> prop to unmount the child after the animation exits.
+            If you need to animate children as they mount and unmount, use a <code>Transition.Group</code>. 
+          </Message.Item>
+          <Message.Item>
+            It is paramount that each child of a <code>Transition.Group</code>
+            has an unique <code>key</code> prop that stays constant through rerenders.
+            In particular, it is likely that <code>keys</code> based on pseudo-random identifiers or simple loop increment variables will lead to unexpected behavior.
```

Comment:
@keksipurkki This is definitely a React issue and is most definitely not unique for this project. In fact, [there are lint rules specifically to make sure you are not using something like an index to set your keys](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-array-index-key.md) because when the DOM updates your keys will not be unique. If they are not always going to be unique then you're not following the documentation. The keys need to be unique in ALL states of your application for the siblings you are referring to.